ledger/accountdb.go

```diff
@@ -2677,11 +2671,23 @@ func (qs *accountsDbQueries) listCreatables(maxIdx basics.CreatableIndex, maxRes
 	return
 }
 
+// sql.go has the following contradictory comments:
+
+// Reference types such as []byte are only valid until the next call to Scan
+// and should not be retained. Their underlying memory is owned by the driver.
+// If retention is necessary, copy their values before the next call to Scan.
+
+// If a dest argument has type *[]byte, Scan saves in that argument a
+// copy of the corresponding data. The copy is owned by the caller and
+// can be modified and held indefinitely. The copy can be avoided by
+// using an argument of type *RawBytes instead; see the documentation
+// for RawBytes for restrictions on its use.
+
 func (qs *accountsDbQueries) lookupKeyValue(key string) (pv persistedKVData, err error) {
 	err = db.Retry(func() error {
-		var v sql.NullString
+		var val []byte
 		// Cast to []byte to avoid interpretation as character string, see note in upsertKvPair
-		err := qs.lookupKvPairStmt.QueryRow([]byte(key)).Scan(&pv.round, &v)
+		err := qs.lookupKvPairStmt.QueryRow([]byte(key)).Scan(&pv.round, &val)
```

Comment:
We need to discuss whether it's legitimate to return this slice without copying.

Reply:
Resolving with an additional comment since @algorandskiy checked the source code.